pandevice/__init__.py

```diff
@@ -136,24 +143,45 @@ def subrelease_num(self):
     def __repr__ (self):
         return "PanOSVersion ('%s')" % str(self)
 
-    def __cmp__ (self, other):
-        if isinstance(other, basestring):
-            other = PanOSVersion(other)
-
-        # Compare subreleases if number part of version is the same
-        if cmp(self.mainrelease, other.mainrelease) == 0:
-            if self.subrelease_type == 'c' and other.subrelease_type != 'c':
-                return -1
-            elif self.subrelease_type == 'b' and other.subrelease is None:
-                return -1
-            elif self.subrelease_type != 'c' and other.subrelease_type == 'c':
-                return 1
-            elif self.subrelease is None and other.subrelease_type == 'b':
-                return 1
-
-        return cmp(self.version, other.version)
-
-
+    def __lt__(self, other):
```

Comment:
The logic is flawed in here somewhere.  This is evaluating to `True` and should be `False`:

`PanOSVersion('6.1.0') < PanOSVersion('6.1.0')`

I think it would be worthwhile to add test cases around this class' changes to make sure that these evaluations are working as expected.  It will probably be a lot of tests, but that's ok.  Since pytest is being used as the test runner now, go ahead and use a parameterized test to make your life easier.